math/minuit/src/TFitter.cxx

```diff
@@ -137,8 +137,12 @@ void TFitter::GetConfidenceIntervals(Int_t n, Int_t ndim, const Double_t *x, Dou
    Double_t c=0;
 
    Double_t *matr = GetCovarianceMatrix();
-   if (!matr)
+   if (!matr){
+      delete [] grad;
```

Comment:
A common pattern in ROOT is to assign the variable nullptr after it was deleted. This avoids crashes on double deletion and use after delete issues.

Adding `grad = nullptr;` should solve this.